src/Interpreters/ITokenExtractor.cpp

```diff
@@ -270,4 +285,19 @@ void SplitTokenExtractor::substringToGinFilter(const char * data, size_t length,
             gin_filter.addTerm(data + token_start, token_len);
 }
 
+#if USE_DATASKETCHES
+void SplitTokenExtractor::stringToSketch(const char * data, size_t length, datasketches::frequent_items_sketch<std::string> & sketch) const
+{
+    size_t cur = 0;
+    size_t token_start = 0;
+    size_t token_len = 0;
+
+    while (cur < length && nextInString(data, length, &cur, &token_start, &token_len))
+    {
+        std::string key(data + token_start, token_len);
+        sketch.update(key);
```

Comment:
I assume, that it doesn't control how often token will be pushed from single granule?
IE, ideally we want to "increment" counter for token only once for each granula to have accurate statistics on how many granules have this particular token

Reply:
"I assume, that it doesn't control how often token will be pushed from single granule?"

Yes, currently we control only total amount of token in all granules and choose items in common filter with this statistic. Your version seems more logical, I'll change it. Maybe with these fixes sketches won't be needed